code/operators/solution/operators.sol.cpp

```diff
@@ -0,0 +1,197 @@
+#include <cassert>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+
+class Fraction {
+ public:
+  explicit Fraction(int i) : m_num(i), m_denom(1) {}
+  Fraction(int num, int denom) : m_num(num), m_denom(denom) {}
+
+  int num() const { return m_num; }
+  int denom() const { return m_denom; }
+
+  void normalize() {
+    const int gcd = std::gcd(m_num, m_denom);
+    m_num /= gcd;
+    m_denom /= gcd;
+  }
+
+  Fraction& operator*=(int i) {
+    m_num *= i;
+    m_denom *= i;
+    return *this;
+  }
+
+  Fraction& operator*=(Fraction f) {
+    m_num *= f.num();
+    m_denom *= f.denom();
+    normalize();
+    return *this;
+  }
+
+  Fraction& operator+=(int i) {
+    m_num += i * m_denom;
+    return *this;
+  }
+
+  Fraction& operator+=(Fraction f) {
+    const int myDenom = m_denom;
+    *this *= f.denom();
```

Comment:
that's calling the operator I've changed above. Indeed the code works, but I would change it to avoid having a public operator*(int) which actually does not multiply the fraction by the int, but rather "denormalize" it

Reply:
Sure, I will adapt it accordingly!